engine/crates/partial-caching/src/planning/variables.rs

```diff
@@ -0,0 +1,59 @@
+use cynic_parser::{
+    executable::{ids::VariableDefinitionId, Argument, Directive, OperationDefinition},
+    ExecutableDocument,
+};
+use indexmap::IndexSet;
+
+use crate::query_subset::CacheGroup;
+
+pub fn variables_required(
+    partition: &CacheGroup,
+    document: &ExecutableDocument,
+    operation: OperationDefinition<'_>,
+) -> IndexSet<VariableDefinitionId> {
+    let mut variables = partition
+        .selections
+        .iter()
+        .flat_map(|id| -> Box<dyn Iterator<Item = &str>> {
+            match document.read(*id) {
+                cynic_parser::executable::Selection::Field(field) => Box::new(
+                    variables_used_in_argument(field.arguments())
+                        .chain(variables_used_in_directive(field.directives())),
```

Comment:
:bulb: I would 100% have forgotten to check for variable usage in directives.

Reply:
Yeah, I nearly did as well